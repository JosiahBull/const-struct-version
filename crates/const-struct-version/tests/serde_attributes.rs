#![allow(dead_code)]

use const_struct_version::StructVersion;

#[derive(StructVersion, serde::Deserialize)]
#[serde(rename_all = "camelCase")]
struct User {
    #[serde(rename = "userId")]
    id: String,
    login_count: u32,
}

#[derive(StructVersion, serde::Deserialize)]
struct User2 {
    #[serde(rename = "userId")]
    id: String,
    login_count: u32,
}

#[derive(StructVersion, serde::Deserialize)]
#[serde(rename_all = "camelCase")]
struct User3 {
    id: String,
    login_count: u32,
}

#[test]
fn main() {
    let version = <User as StructVersion>::version();
    let version2 = <User2 as StructVersion>::version();
    let version3 = <User3 as StructVersion>::version();

    insta::assert_debug_snapshot!(version);
    insta::assert_debug_snapshot!(version2);
    insta::assert_debug_snapshot!(version3);

    // Ensure all are different.
    assert_ne!(version, version2);
    assert_ne!(version, version3);
    assert_ne!(version2, version3);
}
