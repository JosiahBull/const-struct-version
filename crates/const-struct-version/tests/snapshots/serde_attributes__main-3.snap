---
source: crates/const-struct-version/tests/serde_attributes.rs
expression: version3
---
"0cb237abc3f3332b7d3bd0ad97ea92d08caaee72"
